src/js/common/components/CopyTextToClipboardButton/CopyTextToClipboardButton.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+
+import { IconButton, Tooltip } from '@material-ui/core';
+import { Done, FileCopyOutlined } from '@material-ui/icons';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+const CopyTextToClipboardButton = ({ textToCopy }) => {
+  const [isShowingSuccessTooltip, setIsShowingSuccessTooltip] = useState(false);
+
+  const { t } = useTranslation('common');
+
+  const copyToClipboard = () => {
+    navigator.clipboard.writeText(textToCopy);
```

Comment:
It is better to handle asynchronous calls more appropriately.
I would use the [mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText) example as a base
navigator.clipboard.writeText(textToCopy).then(()=> {
    // success message
    }, ()=> {
    // error message
})

and would remove the setTimeout

Reply:
The setTimeout is used to give a visual feedback to the user.
But I agree that async calls should be handled appropriately.